include/inexor/vulkan-renderer/wrapper/device.hpp

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/availability_checks.hpp"
+#include "inexor/vulkan-renderer/settings_decision_maker.hpp"
+
+#include <spdlog/spdlog.h>
+#include <vulkan/vulkan.h>
+
+#include <cassert>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+namespace inexor::vulkan_renderer::wrapper {
+
+/// @brief A RAII wrapper for VkDevice, VkPhysicalDevice and VkQueues.
+class Device {
+private:
+    VkDevice device = VK_NULL_HANDLE;
+    VkPhysicalDevice graphics_card = VK_NULL_HANDLE;
+    VkQueue graphics_queue = VK_NULL_HANDLE;
+    VkQueue present_queue = VK_NULL_HANDLE;
+    VkQueue transfer_queue = VK_NULL_HANDLE;
+    VkSurfaceKHR surface;
+
+    std::optional<std::uint32_t> present_queue_family_index = std::nullopt;
+    std::optional<std::uint32_t> graphics_queue_family_index = std::nullopt;
+    std::optional<std::uint32_t> transfer_queue_family_index = std::nullopt;
+
+    // TODO: Make proper use of queue priorities in the future.
+    const float global_queue_priority = 1.0f;
```

Comment:
This should probably be moved to the source file in an anonymous namespace and made `constexpr`

Reply:
Yes.

Reply:
Done.